### PR TITLE
Fix CUDA batch_norm eval returning populated save_mean/save_invstd

### LINKED ISSUE #85960: Discrepancy in output shape for batch_norm inference mode between CUDA and CPU
### 🐛 Describe the bug

For CPU inputs `torch.native_batch_norm` returns 0-sized outputs, but for CUDA inputs it returns a copy of `running_mean` argument and inverse of standard deviation.
What's the reason for this discrepancy and what's the correct output? Can we align the behavior of two backends?

CPU code path:
https://github.com/pytorch/pytorch/blob/8f4edf1e1dc9419f0bab66a67c8f149d7b53fc25/aten/src/ATen/native/Normalization.cpp#L684-L687

CUDA code path:
https://github.com/pytorch/pytorch/blob/fe87ae692f813934d1a74d000fd1e3b546c27ae2/aten/src/ATen/native/cuda/Normalization.cu#L440-L443

### Versions

Latest master.

### PR DESCRIPTION
**Pull request by jashshah999**
Fixes #85960

## Summary

In eval mode, the CUDA `native_batch_norm` kernel was copying `running_mean` into `save_mean` and computing `save_invstd` from `running_var`, returning populated tensors of shape `[num_features]`. The CPU kernel returns empty (size-0) tensors for both, since they are not needed for the backward pass in eval mode.

```python
import torch
bn = torch.nn.BatchNorm2d(3).eval()
x = torch.randn(2, 3, 4, 4)

# CPU - correct
_, save_mean_cpu, save_invstd_cpu = torch.native_batch_norm(
    x, bn.weight, bn.bias, bn.running_mean, bn.running_var, False, 0.1, 1e-5)
print(save_mean_cpu.shape)    # torch.Size([0])
print(save_invstd_cpu.shape)  # torch.Size([0])

# CUDA - before this fix
_, save_mean_cuda, save_invstd_cuda = torch.native_batch_norm(
    x.cuda(), bn.weight.cuda(), bn.bias.cuda(), bn.running_mean.cuda(),
    bn.running_var.cuda(), False, 0.1, 1e-5)
print(save_mean_cuda.shape)    # torch.Size([3])  <-- should be [0]
print(save_invstd_cuda.shape)  # torch.Size([3])  <-- should be [0]
```

The fix computes the forward pass using `running_mean` and a locally computed `invstd` from `running_var`, then returns empty `save_mean`/`save_invstd` tensors to match CPU behavior.

## Test plan

- Existing `test_compare_cpu_nn_functional_batch_norm` opinfo test should now pass for the eval case (previously CUDA returned different-shaped auxiliary outputs)
- Manual verification: `torch.native_batch_norm(..., training=False)` on CUDA returns size-0 `save_mean`/`save_invstd`

cc @ngimel @albanD @jjsjann123

### CODE REVIEW

**Comment by jashshah999** [comment]
Gentle ping — the CI failures noted earlier are all infra-related (GTest not found, runner exit code 127), not caused by this change. Happy to rebase if that helps retrigger CI.
